go/store/prolly/tree/json_diff_test.go

```diff
@@ -183,43 +249,164 @@ var simpleJsonDiffTests = []jsonDiffTest{
+		{
+			name: "nested insert 1",
+			from: largeObject,
+			to:   set(largeObject, "$.level7.level5", 2),
+			expectedDiffs: []JsonDiff{
+				{
+					Key:  makeJsonPathKey(`level7`, `level5`),
+					From: lookup(largeObject, "$.level7.level5"),
+					To:   &types.JSONDocument{Val: 2},
+					Type: ModifiedDiff,
+				},
+			},
+		},
+		{
+			name: "nested insert 2",
+			from: largeObject,
+			to:   set(largeObject, "$.level7.level4", 1),
+			expectedDiffs: []JsonDiff{
+				{
+					Key:  makeJsonPathKey(`level7`, `level4`),
+					From: lookup(largeObject, "$.level7.level4"),
+					To:   &types.JSONDocument{Val: 1},
+					Type: ModifiedDiff,
+				},
+			},
+		},
+		{
+			name: "convert object to array",
+			from: largeObject,
+			to:   set(largeObject, "$.level7.level6", []interface{}{}),
+			expectedDiffs: []JsonDiff{
+				{
+					Key:  makeJsonPathKey(`level7`, `level6`),
+					From: lookup(largeObject, "$.level7.level6"),
+					To:   &types.JSONDocument{Val: []interface{}{}},
+					Type: ModifiedDiff,
+				},
+			},
+		},
+		{
+			name: "convert array to object",
+			from: set(largeObject, "$.level7.level6", []interface{}{}),
+			to:   largeObject,
+			expectedDiffs: []JsonDiff{
+				{
+					Key:  makeJsonPathKey(`level7`, `level6`),
+					From: &types.JSONDocument{Val: []interface{}{}},
+					To:   lookup(largeObject, "$.level7.level6"),
+					Type: ModifiedDiff,
+				},
+			},
+		},
+	}
+}
+
+// createLargeDocumentForTesting creates a JSON document large enough to be split across multiple chunks.
+// This is useful for testing mutation operations in large documents.
+// Every different possible jsonPathType appears on a chunk boundary, for better test coverage:
+// chunk 0 key: $[6].children[2].children[0].number(endOfValue)
+// chunk 2 key: $[7].children[5].children[4].children[2].children(arrayInitialElement)
+// chunk 5 key: $[8].children[6].children[4].children[3].children[0](startOfValue)
+// chunk 8 key: $[8].children[7].children[6].children[5].children[3].children[2].children[1](objectInitialElement)
+func createLargeDocumentForTesting2(t *testing.T, ctx *sql.Context, ns NodeStore) IndexedJsonDocument {
+	leafDoc := make(map[string]interface{})
+	leafDoc["number"] = float64(1.0)
+	leafDoc["string"] = "dolt"
+	var docExpression sql.Expression = expression.NewLiteral(newIndexedJsonDocumentFromValue(t, ctx, ns, leafDoc), types.JSON)
+	// require.NoError(t, err)
+	var err error
+
+	for level := 0; level < 8; level++ {
+		docExpression, err = json.NewJSONInsert(docExpression, expression.NewLiteral(fmt.Sprintf("$.level%d", level), types.Text), docExpression)
+		require.NoError(t, err)
+	}
+	doc, err := docExpression.Eval(ctx, nil)
+	require.NoError(t, err)
+	return newIndexedJsonDocumentFromValue(t, ctx, ns, doc)
+}
+
 func TestJsonDiff(t *testing.T) {
 	t.Run("simple tests", func(t *testing.T) {
 		runTestBatch(t, simpleJsonDiffTests)
 	})
+	t.Run("large document tests", func(t *testing.T) {
+		runTestBatch(t, makeLargeJsonDiffTests(t))
```

Comment:
how about calling this `largeJsonDiffTests` so that it's more consistent with the other test function, `simpleJsonDiffTests`?